### DISCUSSION TITLE
Custom target path

### DISCUSSION THREAD

**Discussion by amitpundir** [axkibe/lsyncd#699] [Q&A] [answered]
how can i use a custom target path as like
target="/logs/$(/bin/date +\%Y/\%m/\%d)/$(/usr/bin/hostname)/reports/ ?

**Comment by axkibe** [comment, marked as answer]
Questions/requests like these were exactly the reason Lsyncd went from the version 1 implementation which was pure C with XML config files to Lua as "config file" format, which is a full feature programming language, so on this you can do whatever you heart desires.. and is codable.

So this would be:
```Lua
function exec( cmd )
	local f = io.popen( cmd )
	local result = f:read( '*a') or ''
	f:close( )
	result = string.gsub( result, '\n$', '' )
	return result
end

local hostname = exec( '/usr/bin/hostname' )
local date = exec( '/bin/date +%Y/%m/%d' )
local target = '/logs/'..date..'/'..hostname..'/reports/'

print( target )
```
Just instead of printing the target variable (I used just the standalone Lua interpreter here) feed this variable in a Lsyncd sync{} call.

**Reply by poelzi** [reply]
This solution will only work when lsyncd is restarted every day, since the target is not reevaluated. You would also need to overwrite the meta table so target gets reevaluated.
I will add the functionality that, if target is a function, it will be executed and the return value is used. This will allow easy and powerful customization.

**Reply by axkibe** [reply]
That would be nice, you might want to insert a blanket delay when the evaluated target value changes so a full rsync is done on the new destination.

Basically it could also be done on userscript, there is a UserAlarm interface (albeit I think not been documented) that would allow a user script to calculate next midnight time and execute a function at that moment. Alas the biggest issue here is, as far i remember there is no code that allows to remove a sync on the fly (might need some work to be sure not to leave any things in undefined state), so this way a user script cannot remove the old days sync{} and setup a new one for the next day.

Or simply kill and restart Lsyncd after midnight from a cronjob.

**Reply by poelzi** [reply]
There is now a full sync message to retrigger full sync and even a crontab interface to specify recuring full syncs.
